src/client/src/OpenFin/utils/poppedOutSections.ts

```diff
@@ -0,0 +1,16 @@
+export let tilesPoppedOut = false
```

Comment:
Is this way of managing state recommened for OpenFin? If not I think this could do with a refacor, maybe using Contexts or Rx to get the state where it needs to be

Reply:
Yes, the max number of popped out sections should be 2. However, since I'm using view attached to track when sections get popped out, it gets called 3 times on initialization, and when a view gets attached I decrement numPoppedOutSections. And the only reason I am using the viewAttached listener over windowCreated is because we are using the openfin popout button, which automatically assigns "internally-generated-..." as the window name, so I won't know which window was popped out in the windowCreated listener, otherwise I would not need to set the initial value of numPoppedOutSections as 3. Since we know main window and snapshots and contacts are the only windows created on initialization.

Reply:
So yes, if in the openfin upgrade, we are given control over naming the window created from openfin's popout, I will refactor to use windowCreated and this will all make much more sense.